modules/ROOT/pages/java-support.adoc

```diff
@@ -0,0 +1,160 @@
+. In Anypoint Studio, update the connectors in your app to the latest Java 17-compatible version. 
+. xref:mule-runtime::deploying.adoc[Deploy your applications] to a target running on Mule 4.6 or later.
+
+[[custom-connectors]]
+== Custom Connectors
+
+Custom connectors are any connectors that are not developed and maintained by MuleSoft, including connectors that are built by MuleSoft partners or customers. If you are using a custom connector in your app, you must update your connector to run on Java 17 and Mule runtime 4.6 and later. 
+
+* If you are a MuleSoft partner:
++
+. Update your connector that is generated from:
+  ** xref:partner-connector-upgrade.adoc#upgrade-your-mule-sdk-connectors[Mule SDK]
+  ** xref:partner-connector-upgrade.adoc#upgrade-your-rest-sdk-connectors[REST SDK]
+  ** xref:partner-connector-upgrade.adoc#upgrade-your-rest-connect-connectors[REST Connect]
+. Test your connector using xref:partner-connector-upgrade.adoc#test-your-custom-connector-with-mtf[MTF].
+. xref:mule-runtime::deploying.adoc[Deploy your applications] to a target running on Mule 4.6 or later.
+
+* If you are a MuleSoft customer:
++
+. Update your connector that is generated from:
+  ** xref:customer-connector-upgrade.adoc#upgrade-your-mule-sdk-connectors[Mule SDK]
+  ** xref:customer-connector-upgrade.adoc#upgrade-your-rest-connect-connectors[REST Connect]
+. Test your connector using xref:customer-connector-upgrade.adoc#test-your-custom-connector-with-munit[MUnit].
+. xref:mule-runtime::deploying.adoc[Deploy your applications] to a target running on Mule 4.6 or later.
+
+[[policies]]
+== Policies
+
+The MuleSoft-included Mule Gateway policies are compatible with Java 17 beginning with the Mule runtime 4.6 release. These policies continue to have standard support for Java 8 until February 2025, so it's best to start updating your policies as soon as possible. 
+
+[IMPORTANT]
+To ensure your API proxies or Mule apps are protected when upgrading, upgrade your policies *before* upgrading your API proxies or Mule apps.
+
+For details about how to upgrade your policies, see xref:upgrade-policies-proxies.adoc#upgrading-automated-policies[Upgrading Automated Policies] and xref:upgrade-policies-proxies.adoc#upgrading-api-level-policies[Upgrading API-Level Policies].
+
+[[api-proxies]]
+== API Proxies
+
+The MuleSoft-included API proxies are compatible with Java 17 beginning with the Mule runtime 4.6 release. These API proxies continue to have standard support for Java 8 until February 2025, so it’s best to start updating your API proxies as soon as possible. 
+
+The steps to upgrade are a little different, depending on which deployment model you use. 
+
+* If you use Basic endpoint, deploy the adapted application to the server from Mule runtime and connect it to API Manager using autodiscovery. For more information, see xref:mule-gateway::mule-gateway-config-autodiscovery-mule4.adoc[Configuring Mule Gateway API Autodiscovery in a Mule 4 Application].
```

Comment:
First sentence is a little hard to parse. Deploy from Mule to the server?

Do you need the hard-coded link text?